csvcubed/csvcubed/utils/qb/components.py

```diff
@@ -0,0 +1,79 @@
+"""
+Cube Components
+---------------
+
+Utilities to help when handling qube components data.
+"""
+
+from enum import Enum
+from pathlib import Path
+
+
+class ComponentPropertyTypeURI(Enum):
+    """
+    The type uris of component properties..
+    """
+
+    Dimension = "http://purl.org/linked-data/cube#DimensionProperty"
+    """ The component is of type qb:Diemension. """
+
+    Attribute = "http://purl.org/linked-data/cube#AttributeProperty"
+    """ The component is of type qb:Attribute. """
+
+    Measure = "http://purl.org/linked-data/cube#MeasureProperty"
+    """ The component is of type qb:Measure. """
+
+
+class ComponentPropertyType(Enum):
+    """
+    The type of component properties..
+    """
+
+    Dimension = "Dimension"
+    """ The component is of type qb:Diemension. """
+
+    Attribute = "Attribute"
+    """ The component is of type qb:Attribute. """
+
+    Measure = "Measure"
+    """ The component is of type qb:Measure. """
+
+
+def get_printable_component_property_type(property_type: str) -> str:
+    """
+    Produces the user-friendly name of component property type.
+
+    Member of :file:`./utils/qb/components.py`
+
+    :return: `str` - user-friendly name of component property type.
+    """
+    if ComponentPropertyTypeURI.Dimension.value == property_type:
+        return ComponentPropertyType.Dimension.value
+    elif ComponentPropertyTypeURI.Attribute.value == property_type:
+        return ComponentPropertyType.Attribute.value
+    elif ComponentPropertyTypeURI.Measure.value == property_type:
+        return ComponentPropertyType.Measure.value
+    else:
+        raise Exception(f"Property type {property_type} is not supported.")
+
+
+def get_printable_component_property(
+    input_file_path: Path, component_property: str
+) -> str:
+    """
+    Produces the user-friendly property of the component property.
+
+    Member of :file:`./utils/qb/components.py`
+
+    :return: `str` - url or relative path
+    """
+    if component_property.startswith("file://") == False:
+        return component_property
+
+    component_property = component_property.removeprefix("file://")
+    try:
+        component_property_path = Path(component_property)
+        relative_path = component_property_path.relative_to(input_file_path.parent)
```

Comment:
I found that [os.path.relpath](https://docs.python.org/3/library/os.path.html#os.path.relpath) does the relative paths in a more robust way than Pathlib; it does the `../` to backout to parent directory:

```python
>>> os.path.relpath("/home/something", "/chicken/burgers")
'../../home/something'
```